dashboard/src/components/AppView/AppControls/UpgradeButton.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+import { ArrowUpCircle } from "react-feather";
+import { Redirect } from "react-router";
+
+interface IUpgradeButtonProps {
+  updateVersion?: string;
+  upgradeURL: string;
+}
+
+interface IUpgradeButtonState {
+  upgrade: boolean;
+}
+
+class UpgradeButton extends React.Component<IUpgradeButtonProps, IUpgradeButtonState> {
```

Comment:
`react-router` allows you to redirect users programatically. You only need to pass the `history` property from `AppView` to this component and use it to redirect the user:

```jsx
import * as React from "react";
import { ArrowUpCircle } from "react-feather";

interface IUpgradeButtonProps {
  history: any, // You probably want to be more specific
  updateVersion?: string;
  upgradeURL: string;
}
const UpgradeButton = React.SFC<IUpgradeButtonProps> = ({ history, updateVersion, upgradeURL }) => {
  let upgradeButton;

  // Redirect the user
  const onClick = () => history.push(upgradeURL);

  // If the app is outdated highlight the upgrade button
  if (updateVersion) {
    upgradeButton = (
      <div className="tooltip">
        <button className="button upgrade-button" onClick={onClick}>
          <span className="upgrade-text">Upgrade</span>
          <ArrowUpCircle color="white" size={25} fill="#82C341" className="notification" />
        </button>
        <span className="tooltiptext tooltip-top">New version ({updateVersion}) found!</span>
      </div>
    );
  } else {
    upgradeButton = (
      <button className="button" onClick={onClick}>
        Upgrade
      </button>
    );
  }
  return (
    <React.Fragment>
      {upgradeButton}
    </React.Fragment>
  );
}

export default UpgradeButton;
```

Reply:
We've used the `Redirect` container in places to avoid having to pass down a prop all the way. If we do prefer using `push`, I would prefer to just pass down `push` and not the whole `history` prop.